### PR TITLE
revamp python3 version checking

### LINKED ISSUE #7723: 🐞 Bug: crew upgrade aborts
```
chronos@dell-precision-m6700-x86_64 ~/chromebrew $ crew update
... snip ...
Package lists, crew, and library updated.
Checking for package updates...
py3_setuptools could be updated from 62.3.2-1 to 65.6.3-py3.11
py3_wheel could be updated from 0.37.1-1 to 0.38.4-py3.11
py3_pip could be updated from 21.3.1 to 22.3.1-py3.11
itstool could be updated from 2.0.7 to 2.0.7-py3.11
py3_codespell could be updated from 2.2.2 to 2.2.2-py3.11
py3_pyelftools could be updated from 0.28 to 0.29-py3.11
py3_tomli could be updated from 2.0.1-1 to 2.0.1-py3.11
py3_pep517 could be updated from 0.12.0-1 to 0.12.0-py3.11
py3_build could be updated from 0.8.0-1 to 0.9.0-py3.11
py3_installer could be updated from 0.5.1-1 to 0.5.1-py3.11
py3_flit_core could be updated from 3.7.1-1 to 3.8.0-py3.11
meson could be updated from 0.64.1 to 0.64.1-py3.11

Run `crew upgrade` to update all packages or `crew upgrade <package1> [<package2> ...]` to update specific packages.
chronos@dell-precision-m6700-x86_64 ~/chromebrew $ crew upgrade
Updating packages...
py3_setuptools: Setuptools is the python build system from the Python Packaging Authority.
Performing pre-flight checks...
chronos@dell-precision-m6700-x86_64 ~/chromebrew $
```

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #7723 

## Additional information

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=py_constant CREW_TESTING=1 crew update
```

### CODE REVIEW

File: lib/const.rb [hunk 1]
```diff
@@ -295,6 +295,7 @@
 CREW_LIB_SUFFIX = ARCH == 'x86_64' ? '64' : ''
 CREW_CMAKE_LIBSUFFIX_OPTIONS = "#{CREW_CMAKE_OPTIONS} -DLIB_SUFFIX=#{CREW_LIB_SUFFIX}"
 
+PY3_VERSION = File.file?("#{CREW_PREFIX}/bin/python3") ? `#{CREW_PREFIX}/bin/python3 --version | cut -d" " -f2 | cut -d"." -f1,2`.chomp.to_f : 404
```
Comment (supechicken) [inline lib/const.rb:298]: We should avoid adding non-core variables to `const.rb`

**Comment by supechicken** [comment]
It would be great if `crew` can handle this natively :) (thinking something like `depends_on 'python' => ['>=', '3.11']`)

**Comment by satmandu** [comment]
> It would be great if `crew` can handle this natively :) (thinking something like `depends_on 'python' => ['>=', '3.11']`)

Please make this happen!

**Comment by satmandu** [comment]
Closing in favor of https://github.com/chromebrew/chromebrew/pull/7725

**Pull request closed without merging**